spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/it/run-envargs/src/main/java/org/test/SampleApplication.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.test;
+
+public class SampleApplication {
+
+
+	public static void main(String[] args) {
+		new Assert("ENV1").expect("5000");
+		new Assert("ENV2").expect("Some Text");
+		new Assert("ENV3").expect("");
+		new Assert("ENV4").expect("");
+
+		System.out.println("I haz been run");
+
+	}
+
+
+	private static class Assert {
```

Comment:
This is a strange way to do this. Why not a method with 2 parameters?

Reply:
First of all, thank you for your comment.
The reason why I did this is encapsulation `System.getenv('key')` and just do it similar to
the `Assertions.assertThat(envKey).isEqualTo(val)`.
Of course, it could be 

```java
void assert(String envKey, String expectedValue) {
 ///
}
```
But in this particular case, it doesn't matter, but still if you have any concerns we can discuss this.